pool.go

```diff
@@ -498,13 +518,21 @@ func (p *Pool) doOverflowDrain() {
 
 func (p *Pool) getExisting() (*ioErrConn, error) {
 	// Fast-path if the pool is not empty. Return error if pool has been closed.
-	select {
-	case ioc, ok := <-p.pool:
-		if !ok {
-			return nil, errClientClosed
+	for {
+		select {
+		case ioc, ok := <-p.pool:
+			if !ok {
+				return nil, errClientClosed
+			}
+			if ioc.expired(p.opts.maxLifetime) {
+				ioc.Close()
+				p.traceConnClosed(trace.PoolConnClosedReasonPoolFull)
```

Comment:
We should add a new trace reason, `trace.PoolConnClosedReasonMaxLifetime`, this current trace isn't accurate (and the one in the below for loop as well).